lib/autoload/course/assessment/programming_test_case_report.rb

```diff
@@ -171,6 +202,22 @@ def passed?
       !failed? && !skipped? && !errored?
     end
 
+    # Combines the different strings above into a hash
+    #
+    # @return [Hash]
+    def messages
+      @messages ||= {
+        'error': error_message,
+        'error_contents': error_contents,
+        'hint': hint,
+        'failure': failure_message,
+        'failure_contents': failure_contents,
+        'output': output
+      }
+      # prune empty and nil values
+      @messages.reject! { |_, v| v.blank? }
```

Comment:
Just fixed. Will run another round of manual testing in the morning. Meanwhile can you eyeball this?